hbase-procedure/src/main/java/org/apache/hadoop/hbase/procedure2/ProcedureExecutor.java

```diff
@@ -1596,9 +1596,11 @@ private LockState executeRollback(Procedure<TEnvironment> proc) {
     } catch (InterruptedException e) {
       handleInterruptedException(proc, e);
       return LockState.LOCK_YIELD_WAIT;
+    } catch (UnsupportedOperationException e) {
+      LOG.debug("Rollback is not supported for {}", proc);
```

Comment:
`UnsupportedOperationException` can be anything else other than rollback step for procedure?

Reply:
Or maybe we can check for `UnsupportedOperationException`'s cause to verify this is rollback only and not something else? Anyways, I don't have any strong case here, just trying to understand it better.